src/CashuWallet.ts

```diff
@@ -410,47 +484,65 @@ class CashuWallet {
 	/**
 	 * Creates blinded messages for a given amount
 	 * @param amount amount to create blinded messages for
+	 * @param preference optional preference for splitting proofs into specific amounts. overrides amount param
+	 * @param count? optionally set count to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
 	 * @returns blinded messages, secrets, rs, and amounts
 	 */
 	private createRandomBlindedMessages(
 		amount: number,
-		amountPreference?: Array<AmountPreference>
+		amountPreference?: Array<AmountPreference>,
+		count?: number
+	): BlindedMessageData & { amounts: Array<number> } {
+		const amounts = splitAmount(amount, amountPreference);
+		return this.createBlindedMessages(amounts, count);
+	}
+
+	/**
+	 * Creates blinded messages for a according to @param amounts
+	 * @param amount array of amounts to create blinded messages for
+	 * @param count? optionally set count to derive secret deterministically. CashuWallet class must be initialized with seed phrase to take effect
+	 * @returns blinded messages, secrets, rs, and amounts
+	 */
+	private createBlindedMessages(
+		amounts: Array<number>,
+		count?: number,
+		keysetId?: string
 	): BlindedMessageData & { amounts: Array<number> } {
 		const blindedMessages: Array<SerializedBlindedMessage> = [];
 		const secrets: Array<Uint8Array> = [];
 		const rs: Array<bigint> = [];
-		const amounts = splitAmount(amount, amountPreference);
 		for (let i = 0; i < amounts.length; i++) {
-			const secret = randomBytes(32);
+			let deterministicR = undefined;
+			let secret = undefined;
+			if (this._seed && count != undefined) {
+				secret = deriveSecret(this._seed, keysetId??this.keysetId, count + i);
+				deterministicR = bytesToNumber(deriveBlindingFactor(this._seed, keysetId??this.keysetId, count + i));
+			} else {
+				secret = randomBytes(32);
+			}
 			secrets.push(secret);
-			const { B_, r } = dhke.blindMessage(secret);
+			const { B_, r } = dhke.blindMessage(secret, deterministicR);
 			rs.push(r);
 			const blindedMessage = new BlindedMessage(amounts[i], B_);
 			blindedMessages.push(blindedMessage.getSerializedBlindedMessage());
 		}
-		return { blindedMessages, secrets, rs, amounts };
+		return { amounts, blindedMessages, rs, secrets };
```

Comment:
Why is this re-arranged? Does it have a meaningful impact on anything or just formatting?

Reply:
lol, I have no explanation for this. Maybe my OCD made me order them alphabetically.

Reply:
```suggestion
		return { blindedMessages, secrets, rs, amounts };
```